students/alexander_boone/lesson02/assignment/code/charges_calc.py

```diff
@@ -5,43 +5,135 @@
 import json
 import datetime
 import math
+import logging
+
+
+def set_logging_settings(level):
+    '''Setup logging settings based on level input to command line.'''
+
+    # 0 (Default) - no debug messages or log file
+    if level == '0':
+        logging.disable()
+        return 0
+
+    log_format = '%(asctime)s %(filename)s:%(lineno)-3d %(levelname)s %(message)s'
+    formatter = logging.Formatter(log_format)
+    log_file = datetime.datetime.now().strftime('%Y-%m-%d') + '.log'
+
+    file_handler = logging.FileHandler(log_file)
+    file_handler.setFormatter(formatter)
+    console_handler = logging.StreamHandler()
+    console_handler.setFormatter(formatter)
+    logger = logging.getLogger()
+
+    # 1 - only error messages
+    if level == '1':     
+        file_handler.setLevel(logging.ERROR)
+        console_handler.setLevel(logging.ERROR)
+        
+        logger.setLevel(logging.ERROR)
+        logger.addHandler(file_handler)
+        logger.addHandler(console_handler)
+
+    # 2 - Error messages and warnings
+    if level == '2':
+        file_handler.setLevel(logging.WARNING)
+        console_handler.setLevel(logging.WARNING)
+
+        logger = logging.getLogger()
+        logger.setLevel(logging.WARNING)
+        logger.addHandler(file_handler)
+        logger.addHandler(console_handler)
+
+    # 3 - Error messages, warnings, and debug messages
+    if level == '3':
+        file_handler.setLevel(logging.WARNING)
+        console_handler.setLevel(logging.DEBUG)
+        
+        logger = logging.getLogger()
+        logger.setLevel(logging.DEBUG)
+        logger.addHandler(file_handler)
+        logger.addHandler(console_handler)
+    
```

Comment:
Make sure you capture the case in which an invalid level is entered, it should cause your program to error out.